.github/workflows/test.yml

```diff
@@ -0,0 +1,37 @@
+name: Test source code
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+      - run: npm ci
+      - run: npm run lint
+  test:
+    environment: ${{ github.event_name == 'pull_request' && 'external' || 'internal' }}
```

Comment:
Why this check?
This means this is run with external PRs and anything internal right?

Reply:
This means that we only have to approve acceptance tests on pull requests; checks on `master` run automatically.

Reply:
It was a bit contrived; I've simplified it with GitHub Environments and branch protection in mind.